analyses/cnv-chrom-plot/util/bin-coverage.R

```diff
@@ -0,0 +1,160 @@
+  #   seg_ranges: A GenomicRanges object that contains a `status` and a `biospecimen` slot.  
+  #               The `biospecimen slot will be used to split out the `sample_id`'s corresponding ranges.   
+  #               The `status` slot should have gain/loss/neutral. 
+  #   bin_ranges: A binned GenomicRanges made from tileGenome that has been uncompressed with `unlist`. 
+  #   frac_threshold: What coverage fraction do we need to make the call?
+  #   uncallable_threshold: What fraction of a bin needs to be callable for us
+  #                         us to make a status call?
+  #
+  # Returns:
+  #  a small data.frame that contains the status call of the sample for each bin. 
+  #
+  # Extract the ranges for this sample
+  sample_seg_ranges <- seg_ranges[which(seg_ranges$biospecimen == sample_id)]
+  
+  # Split ranges into their respective statuses
+  gain_ranges <- sample_seg_ranges[sample_seg_ranges$status == "gain"]
+  loss_ranges <- sample_seg_ranges[sample_seg_ranges$status == "loss"]
+  neutral_ranges <- sample_seg_ranges[sample_seg_ranges$status == "neutral"]
+  
+  # Calculate length of each type of status per bin
+  gain_per_bin <- bp_per_bin(bin_ranges, gain_ranges)
+  loss_per_bin <- bp_per_bin(bin_ranges, loss_ranges)
+  neutral_per_bin <- bp_per_bin(bin_ranges, neutral_ranges)
+  uncallable_per_bin <- bp_per_bin(bin_ranges, uncallable_ranges)
+    
+  # Format this data into one data.frame where each row is a bin
+  bin_bp_status <- data.frame(
+    bin = as.numeric(1:length(bin_ranges)),
+    # Keep bin width
+    bin_width = bin_ranges@ranges@width
+  ) %>%
+  # Join loss coverage data
+  dplyr::left_join(gain_per_bin,
+                   by = "bin"
+  ) %>%
+  # Rename as .gain
+  dplyr::rename(bp_per_bin.gain = bp_per_bin) %>%
+  # Join loss coverage data
+  dplyr::left_join(loss_per_bin,
+                   by = "bin"
+  ) %>%
+  # Rename as .loss
+  dplyr::rename(bp_per_bin.loss = bp_per_bin) %>%
+  # Join neutral coverage data
+  dplyr::left_join(neutral_per_bin,
+                   by = "bin"
+  ) %>%
+  # Rename as .neutral
+  dplyr::rename(bp_per_bin.neutral = bp_per_bin) %>%
+  # Join uncallable loss coverage data
+  dplyr::left_join(uncallable_per_bin,
+                     by = "bin"
+                   ) %>%
+  # Rename as .gain
+  dplyr::rename(bp_per_bin.uncallable = bp_per_bin) %>% 
+    # If there is an NA, at this point we can assume it means 0
+    dplyr::mutate_at(
+      dplyr::vars(
+        dplyr::starts_with("bp_per_bin")
+      ),
+      ~ tidyr::replace_na(., 0)
+    ) %>%
+    # Calculate the bins percentage of each status
+    dplyr::mutate(
+      frac_gain = bp_per_bin.gain / bin_width,
+      frac_loss = bp_per_bin.loss / bin_width,
+      frac_neutral = bp_per_bin.neutral / bin_width,
+      frac_uncallable = bp_per_bin.uncallable / bin_width
+    ) %>%
+    # Use these percentages for declaring final call per bin based on
+    # the frac_delta_threshold
+    dplyr::mutate(
```

Comment:
I've moved to a not using `NA`s strategy. But do we feel okay about what this leaves or doesn't leaves as `neutral`?